hudi-io/src/main/java/org/apache/hudi/storage/HoodieStorage.java

```diff
@@ -426,6 +426,26 @@ public List<StoragePathInfo> listDirectEntries(List<StoragePath> pathList) throw
     return result;
   }
 
+  /**
+   * Lists the file info of the direct files/directories in the given list of paths
+   * and filters the results, if the paths are directory.
+   *
+   * @param pathList given path list.
+   * @param filter filter to apply.
+   * @return the list of path info of the files/directories in the given paths.
+   * @throws FileNotFoundException when the path does not exist.
+   * @throws IOException           IO error.
+   */
+  @PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)
+  public List<StoragePathInfo> listDirectEntries(List<StoragePath> pathList,
+                                                 StoragePathFilter filter) throws IOException {
+    List<StoragePathInfo> result = new ArrayList<>();
+    for (StoragePath path : pathList) {
+      result.addAll(listDirectEntries(path, filter));
+    }
+    return result;
```

Comment:
use `pathList.stream().flatMap...`

Reply:
We cannot use stream here because the lambda `listDirectEntries` need to throw `IOException`

Reply:
Got it, makes sense.